### ISSUE TITLE
`reportUnusedParameter` false positive on abstract setter

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#605]
Code sample in [basedpyright playground](https://basedpyright.com/?pythonVersion=3.13&typeCheckingMode=all&code=GYJw9gtgBAhgRgYygSwgBzCALlAggIQGEAaWOAZyxBgSwgFMsALMAEwChOEAbGc8qPhggAFAUIBKAFzsocqAAE04NPWwBPWfIXxK1Wg2Zstc1vWBRgYMCPL1uwCVAC0APih6pAOh8nFVsC87LCw1PzMLANt7YFIAO3oAdwB9ADcYbgBXeikPKmkoAGIoEHoMbABVOMy7VgAFYRhDMPlWqB8vdiA)

```python
from abc import ABC, abstractmethod


class Bar(ABC):
    @property
    @abstractmethod
    def foo(self) -> str:...
    @foo.setter
    def foo(self, new_value: str): # reportUnusedParameter
        ...

```
related: #604
